### PR TITLE
migrate-to-Angular-11

### PR DESCRIPTION
**Pull request by yhedholm**
About half of the changes were 'automatic'. (You recognize them because they have spaces in the input brackets.)

The other half were related to removing all references to isNullOrUndefined(), updating some types, and changing some imports away from CommonJS modules and toward ECMAScript modules.

I had to manually update the Bootstrap hookup and did not migrate @angular/material from 10 to 11 because it tried to migrate  the whole project to Angular12, which somehow was out of sync and wouldn't run.

### CODE REVIEW

File: frontend/rolecall/src/app/app/site_header.component.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {Component, ElementRef, Input, OnInit, ViewChild} from '@angular/core';
-import {isNullOrUndefined} from 'util';
+//import {isNullOrUndefined} from 'util';
```
Comment (mikzat) [inline frontend/rolecall/src/app/app/site_header.component.ts:2]: remove

File: frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import {Pipe, PipeTransform} from '@angular/core';
-import {isNullOrUndefined} from 'util';
+//import {isNullOrUndefined} from 'util';
```
Comment (mikzat) [inline frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts:2]: remove

File: frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
 @Pipe({
   name: 'emptyStringIfUndefined'
 })
 export class EmptyStringIfUndefinedPipe implements PipeTransform {
 
   public transform(value: unknown): unknown {
-    if (isNullOrUndefined(value)) {
+    if (!value) {
```
Comment (mikzat) [inline frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts:10]: Shouldn't this be `!!`? I think currently this change caused the opposite logic.

File: frontend/rolecall/src/app/unavailability/unavailability-editor.component.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import {Component, OnInit} from '@angular/core';
 import {MatDatepickerInputEvent} from '@angular/material/datepicker';
 import {MatSelectChange} from '@angular/material/select';
-import {isNullOrUndefined} from 'util';
+//import {isNullOrUndefined} from 'util';
```
Comment (mikzat) [inline frontend/rolecall/src/app/unavailability/unavailability-editor.component.ts:4]: remove

File: frontend/rolecall/src/app/mocks/mock_user_backend.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import {HttpResponse} from '@angular/common/http';
 import * as APITypes from 'src/api_types';
-import {isNullOrUndefined} from 'util';
+//mport {isNullOrUndefined} from 'util';
```
Comment (mikzat) [inline frontend/rolecall/src/app/mocks/mock_user_backend.ts:3]: remove

**Review by mikzat — CHANGES_REQUESTED** [review]

File: frontend/rolecall/src/app/api/user_api.service.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import {EventEmitter, Injectable} from '@angular/core';
 import * as moment from 'moment';
 import * as APITypes from 'src/api_types';
 import {environment} from 'src/environments/environment';
-import {isNullOrUndefined} from 'util';
+//import {isNullOrUndefined} from 'util';
```
[the remark this answers is not in the document]
Comment (yhedholm) [reply frontend/rolecall/src/app/api/user_api.service.ts:6]: Done.

File: frontend/rolecall/src/app/app/site_header.component.ts (hunk 1 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/app/site_header.component.ts:2]: Done.

File: frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts (hunk 2 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts:2]: Done.

File: frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts (hunk 3 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts:10]: I think it is OK.
Originally, if value was null, the if statement would execute.
The new code, if value is null, the if statement will execute.

File: frontend/rolecall/src/app/unavailability/unavailability-editor.component.ts (hunk 4 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/unavailability/unavailability-editor.component.ts:4]: Done.

File: frontend/rolecall/src/app/mocks/mock_user_backend.ts (hunk 5 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/mocks/mock_user_backend.ts:3]: Done.

File: frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts (hunk 3 above)
Comment (mikzat) [reply frontend/rolecall/src/app/common_components/empty_string_if_undefined.pipe.ts:9]: Ah got it.

**Review by mikzat — APPROVED** [review]

**Pull request merged**